src/Workspaces/Remote/ServiceHub/Shared/RoslynJsonConverter.RoslynOnly.cs

```diff
@@ -79,5 +81,34 @@ protected override void WriteValue(JsonWriter writer, TodoComment todoComment, J
                 writer.WriteEndObject();
             }
         }
+
+        private class PackageSourceJsonConverter : BaseJsonConverter<PackageSource>
+        {
+            protected override PackageSource ReadValue(JsonReader reader, JsonSerializer serializer)
+            {
+                Contract.ThrowIfFalse(reader.TokenType == JsonToken.StartObject);
+
+                var name = ReadProperty<string>(reader);
```

Comment:
❔ Does this just assume `name` and `source` appear in that order? If so, it looks like it will work but it's a pretty fragile way to handle JSON content (makes an assumption that content creators typically feel safe ignoring).